### ISSUE TITLE
Problems following "development of this repository" 

### ISSUE THREAD

**Issue by jbcpollak** [rospypi/simple#20]
Hello,

I am excited to use this repo, its so much better than having to use catkin or klunking around with rosbridge and roslibpy.

I wanted to make sure I was able to compile the code myself, but when I tried to follow the directions I realized two things:

1) They don't mention you need to run `git submodule update --init --recursive` first - I wanted to create a PR to add it to README.md, but I'm not exactly sure where it should go.

2) Once past that, I ran `rospy-build build`, but got this error:

```
creating dist
Creating tar archive
removing 'cv_bridge-1.13.0.post0' (and everything under it)
copy
usage:  [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or:  --help [cmd1 cmd2 ...]
   or:  --help-commands
   or:  cmd --help

error: invalid command 'bdist_wheel'
```

Any thoughts on that? Also, I'm just wondering, how does this project track different ros-releases, melodic/noetic/etc?

Thanks, looking forward to using this project!

**Comment by otamachan** [comment]
Thanks for your feedback.
I fixed the README somehow.

But from your error message, it seems that `setuptools` is not installed.
Could you provide me the information about your development environment(OS, Python version)?

> Also, I'm just wondering, how does this project track different ros-releases, melodic/noetic/etc?

Actually it does not track the ros releases. All the packages are built from source(github) specified by this [yaml file](https://github.com/rospypi/simple/blob/master/packages.yaml). I keep it to point the latest versions as much as possible, but it's independent of ros releases.

**Comment by jbcpollak** [comment]
Hi, sorry for the delay. We use pyenv to install our Python versions, and typically use Python 3.7.8 on Ubuntu Linux 18.04 or 20.04. I do not have setuptools installed, is that a requirement?

**Comment by otamachan** [comment]
Following steps from `git clone` in a clean directory work for me (Ubuntu18.04, pyenv 1.2.15).
https://github.com/rospypi/simple#development
Could you try this and copy/paste the errors again?

```bash
git clone https://github.com/rospypi/simple.git
cd simple
git submodule update --init --recursive
pyenv local 3.7.5
virtualenv -p python3 dev
. ./dev/bin/activate
pip install --extra-index-url https://rospypi.github.io/simple/ -e rospy-builder/
rospy-build build -d any
rospy-build build -d linux --native
rospy-build index --local any --local linux index
```

**Comment by jbcpollak** [comment]
ok, thanks, I found the problem. We don't use virtualenv, we use Python's built-in venv module. 

this: `python -m venv dev` will result in the error I posted about above. When I switch to the `virtualenv -p python3 dev` command, it worked fine. I didn't have virtualenv installed, so I needed to add it. When I installed virtualenv, it worked fine.

Digging deeper, I found I could make it work with `python -m venv dev`, I think by uninstalling wheel and then reinstalling it _with cache disabled_:

```
pip uninstall wheel
pip install --upgrade --no-cache-dir wheel
```

So I guess unrelated to the repo, but its working now, thanks!
